src/main/java/org/mifos/connector/gsma/account/AccountRoutes.java

```diff
@@ -134,21 +133,76 @@ public void configure() throws Exception {
         from("direct:account-route")
                 .id("account-route")
                 .log(LoggingLevel.INFO, "Getting ${exchangeProperty."+ACCOUNT_ACTION+"} for Identifier")
-                .process(exchange -> {
-                    TransactionChannelRequestDTO channelRequest = objectMapper.readValue(exchange.getProperty(CHANNEL_REQUEST, String.class), TransactionChannelRequestDTO.class);
-                    exchange.setProperty(IDENTIFIER_TYPE, channelRequest.getPayee().getPartyIdInfo().getPartyIdType().toString().toLowerCase());
-                    exchange.setProperty(IDENTIFIER, channelRequest.getPayee().getPartyIdInfo().getPartyIdentifier());
-                })
                 .to("direct:get-access-token")
                 .process(exchange -> exchange.setProperty(ACCESS_TOKEN, accessTokenStore.getAccessToken()))
                 .log(LoggingLevel.INFO, "Got access token, moving on to API call.")
                 .to("direct:get-account-details")
                 .log(LoggingLevel.INFO, "Completed ${exchangeProperty."+ACCOUNT_ACTION+"} ${body}")
                 .choice()
-                    .when(header("CamelHttpResponseCode").isEqualTo("200"))
-                        .to("direct:account-success")
-                    .otherwise()
-                        .to("direct:account-error");
+                .when(exchange -> exchange.getProperty(IS_API_CALL, String.class).equals("true"))
+                    .log(LoggingLevel.INFO, "Setting off API response")
```

Comment:
@SubhamPramanik : What is the purpose of introducing this variable?

Reply:
Right so, there are use case when we want to send API response (Account management API) and sometimes we to publish it as a message (during a BPMN workflow)